include/aws/common/task_scheduler.h

```diff
@@ -0,0 +1,101 @@
+#ifndef AWS_TASK_SCHEDULER_H
+#define AWS_TASK_SCHEDULER_H
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+#include <aws/common/priority_queue.h>
+#include <stdint.h>
+
+typedef void *aws_task_arg;
+
+/**
+ * Pointer to the scheduled function
+ */
+typedef void(*aws_task_fn)(aws_task_arg arg);
+
+/**
+ * Monotonic clock function
+ */
+typedef int(*aws_task_scheduler_clock)(uint64_t *);
+
+struct aws_task {
+    aws_task_fn fn;
+    aws_task_arg arg;
+};
+
+struct aws_task_scheduler {
+    struct aws_allocator *alloc;
+    struct aws_priority_queue queue;
+    uint64_t owning_thread;
+    aws_task_scheduler_clock clock;
+};
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    /**
+     * Initializes a task scheduler instance with a clock function and config. A priority queue will be allocated 
+     * in dynamic mode with initial size of 7
```

Comment:
Don't document initial sizes and other such implementation details in the .h file. These are not part of your API contract.